torchaudio/datasets/musdb_hq.py

```diff
@@ -0,0 +1,93 @@
+import os
+from pathlib import Path
+from typing import Tuple, Union, List, Optional
+
+import torch
+import torchaudio
+from torch import Tensor
+from torch.utils.data import Dataset
+from torchaudio import transforms as T
+
+_URL = "https://zenodo.org/record/3338373/files/musdb18hq.zip"
+_EXT = ".wav"
+
+
+class MUSDB_HQ(Dataset):
+    def __init__(
+        self,
+        root: Union[str, Path],
+        sources: List[str],
+        subset: str,
+        samplerate: int = 44100,
+        split: Optional[str] = None,
+        validation: Optional[List[str]] = None,
+    ) -> None:
+        self.sources = sources
+        self.user_samplerate = samplerate
+        self.split = split
+        self.validation = validation
+
+        basename = os.path.basename(_URL)
+        basename = basename.rsplit(".", 2)[0]
+        assert subset in ["test", "train"], "`subset` must be one of ['test', 'train']"
+        assert self.split is None or self.split in [
+            "train",
+            "validation",
+        ], "`split` must be one of ['train', 'validation']"
+        assert (
+            self.split is None and self.validation is None or self.split and self.validation
+        ), "`split` and `validation` must both be None or not be None"
+        self._path = os.path.join(root, basename)
+        self._path = os.path.join(self._path, subset)
+        self.names = self._collect_songs()
+
+    def _get_file(self, name, source):
+        return Path(self._path) / name / f"{source}{_EXT}"
+
+    def _load_sample(self, n: int) -> Tuple[torch.Tensor, int, str]:
+        name = self.names[n]
+        wavs = []
+        sample_rate = None
+        for source in self.sources:
+            file = self._get_file(name, source)
+            wav, new_samplerate = torchaudio.load(str(file))
+            if sample_rate is None:
```

Comment:
@nateanl the version in demucs also checked for track_length when loading metadata, do you think this is something we should include as well in the return?

Reply:
I thin so. And to make it generalized to all datasets, I have discussed with @leo19941227 and @xiaohui-zhang about supporting `skip_audio` argument in the dataset, which allows users to fetch the metadata (audio length, text transcription, etc) with faster speed.

@leo19941227 Can you open an issue about how you want to fetch the metadata, then we can share the discussion and code design there? Thanks!

Reply:
Oh sure! No problem, will do that soon.